inst/include/model_simulation.hpp

```diff
@@ -131,5 +146,213 @@ void run_new_infections(int hiv_step,
+        }
+
+        // ART dropout
+        if (pars.art_dropout(time_step) > 0) {
+          for (int hu = 0; hu < pars.treatment_stages; hu++) {
+            intermediate.grad(hm, ha, g) += pars.art_dropout(time_step) * state_next.art_strat_adult(hu, hm, ha, g);
+            intermediate.gradART(hu, hm, ha, g) -=
+                pars.art_dropout(time_step) * state_next.art_strat_adult(hu, hm, ha, g);
+          }
+        }
+      }
+    }
+  }
+}
+
+template<typename real_type>
+void run_art_initiation(int hiv_step,
+                        int time_step,
+                        const Parameters<real_type> &pars,
+                        const State<real_type> &state_curr,
+                        State<real_type> &state_next,
+                        IntermediateData<real_type> &intermediate) {
+  for (int g = 0; g < pars.num_genders; ++g) {
+    intermediate.Xart_15plus = 0.0;
+    intermediate.Xartelig_15plus = 0.0;
+    intermediate.expect_mort_artelig15plus = 0.0;
+    for (int ha = pars.hIDX_15PLUS; ha < pars.age_groups_hiv; ++ha) {
+      for (int hm = intermediate.everARTelig_idx; hm < pars.disease_stages; ++hm) {
+        if (hm >= intermediate.anyelig_idx) {
+          // TODO: Implement special population ART eligibility
+          real_type prop_elig = 1.0;
+          intermediate.artelig_hahm(hm, ha - pars.hIDX_15PLUS) = prop_elig * state_next.hiv_strat_adult(hm, ha, g);
+          intermediate.Xartelig_15plus += intermediate.artelig_hahm(hm, ha - pars.hIDX_15PLUS);
+          intermediate.expect_mort_artelig15plus +=
+              pars.cd4_mortality(hm, ha, g) * intermediate.artelig_hahm(hm, ha - pars.hIDX_15PLUS);
+        }
+        for (int hu = 0; hu < pars.treatment_stages; ++hu)
+          intermediate.Xart_15plus +=
+              state_next.art_strat_adult(hu, hm, ha, g) + pars.dt * intermediate.gradART(hu, hm, ha, g);
+      }
+    }
+
+    // calculate number on ART at end of ts, based on number or percent
+    if (pars.dt * (hiv_step + 1) < 0.5) {
+      if ((!pars.art15plus_isperc(g, time_step - 2)) & (!pars.art15plus_isperc(g, time_step - 1))) {
+        // Both values are numbers
+        intermediate.artnum_hts =
+            (0.5 - pars.dt * (hiv_step + 1)) * pars.art15plus_num(g, time_step - 2) +
+            (pars.dt * (hiv_step + 1) + 0.5) * pars.art15plus_num(g, time_step - 1);
+      } else if (pars.art15plus_isperc(g, time_step - 2) & pars.art15plus_isperc(g, time_step - 1)) {
+        // Both values are percentages
+        intermediate.artcov_hts =
+            (0.5 - pars.dt * (hiv_step + 1)) * pars.art15plus_num(g, time_step - 2) +
+            (pars.dt * (hiv_step + 1) + 0.5) * pars.art15plus_num(g, time_step - 1);
+        intermediate.artnum_hts = intermediate.artcov_hts * (intermediate.Xart_15plus + intermediate.Xartelig_15plus);
+      } else if ((!pars.art15plus_isperc(g, time_step - 2)) & pars.art15plus_isperc(g, time_step - 1)) {
```

Comment:
These parens, and on the line above, are not needed (also elsewhere)

```suggestion
      } else if (!pars.art15plus_isperc(g, time_step - 2) && pars.art15plus_isperc(g, time_step - 1)) {
```